### ISSUE TITLE
if statements in arg lists get evaluated out of order

### ISSUE THREAD

**Issue by dunhamsteve** [sctb/lumen#205]
When an `if` occurs in an argument list, it is evaluated before any proceeding arguments.  I was expecting the arguments to be evaluated in order. 

E.g. I expect the following to evaluate to `(1 2)`, but it evaluates to `(2 1)` (here `str` is a placeholder for a function of two arguments and `add` represents something with side-effects).

```
(with r () (str (add r 1) (if true (add r 2))))
```

**Comment by sctb** [comment]
Yes, you're absolutely right. I've dealt with this behavior in other projects but haven't updated Lumen. Perhaps @shawwn has (or has had) a patch for this?

**Comment by shawwn** [comment]
It's worth fixing. I've run into this too.

Patch soon.
